lib/src/shelf_run.dart

```diff
@@ -38,26 +40,46 @@ Future<ShelfRunContext> shelfRun(
 
   if (useHotReload) {
     withHotreload(() async {
-      final server = await _createServer(
+      try {
+        final server = await _createServer(
+          init,
+          defaultBindPort: defaultBindPort,
+          defaultBindAddress: defaultBindAddress,
+          defaultShared: defaultShared,
+          securityContext: securityContext,
+          onStarted: onStarted,
+        );
+        context._server = server;
+      }
+      on SocketException catch(e){
```

Comment:
Do you think it's good to raise any exception, to be not only limited to `SocketException`? What you think?